chain.go

```diff
@@ -166,14 +172,25 @@ func (c *chain) replace(name string, args ...interface{}) {
 // Remove last element from chain path.
 func (c *chain) leave() {
 	if len(c.context.Path) == 0 {
-		panic("unpaired enter/leave")
+		panic("unpaired enter/leave on path")
+	}
+
+	if len(c.context.AliasedPath) == 0 {
+		panic("unpaired enter/leave on aliased path")
 	}
 
 	if !c.failBit {
 		c.handler.Success(&c.context)
 	}
 
 	c.context.Path = c.context.Path[:len(c.context.Path)-1]
+	c.context.AliasedPath = c.context.AliasedPath[:len(c.context.AliasedPath)-1]
+}
+
+// Initialize and set name to aliased path.
+func (c *chain) setAlias(name string) {
+	c.context.AliasedPath = nil
+	c.context.AliasedPath = append(c.context.AliasedPath, name)
```

Comment:
Nitpick: this can be simplified to:

```
c.context.AliasedPath = []string{name}
```

Reply:
Also, chain now has a mutex, so please add lock/unlock to this method too.

Reply:
I fixed on 534ab72 .